community/bolt/src/main/java/org/neo4j/bolt/BoltKernelExtension.java

```diff
@@ -133,61 +141,58 @@ public Lifecycle newInstance( KernelContext context, Dependencies dependencies )
     {
         final Config config = dependencies.config();
         final GraphDatabaseService gdb = dependencies.db();
-        final GraphDatabaseFacade api = (GraphDatabaseFacade) gdb;
-        final LogService logging = dependencies.logService();
-        final Log log = logging.getInternalLog( Sessions.class );
+        final GraphDatabaseAPI api = (GraphDatabaseAPI) gdb;
+        final LogService logService = dependencies.logService();
+        final Log log = logService.getInternalLog( Sessions.class );
 
         final LifeSupport life = new LifeSupport();
 
         final JobScheduler scheduler = dependencies.scheduler();
 
-        Netty4LogBridge.setLogProvider( logging.getInternalLogProvider() );
-
-        Sessions sessions =
-                new MonitoredSessions( dependencies.monitors(),
-                        new ThreadedSessions(
-                                life.add( new StandardSessions( api, dependencies.usageData(), logging,
-                                        dependencies.txBridge() ) ),
-                                scheduler, logging ), Clock.systemUTC() );
-
-        List<ProtocolInitializer> connectors = config
-                .view( enumerate( GraphDatabaseSettings.Connector.class ) )
-                .map( BoltConnector::new )
-                .filter( ( connConfig ) -> BOLT.equals( config.get( connConfig.type ) )
-                        && config.get( connConfig.enabled ) )
-                .map( ( connConfig ) -> {
+        Netty4LogBridge.setLogProvider( logService.getInternalLogProvider() );
+
+        Authentication authentication = authentication( dependencies.config(), dependencies.authManager(), logService );
+        StandardSessions standardSessions = life.add(
+                new StandardSessions( api, dependencies.usageData(), logService, dependencies.txBridge(),
+                        authentication, dependencies.neoStoreDataSource() ) );
+        ThreadedSessions threadedSessions = new ThreadedSessions( standardSessions, scheduler, logService );
+        Sessions sessions = new MonitoredSessions( dependencies.monitors(), threadedSessions, Clock.systemUTC() );
+
+        List<ProtocolInitializer> connectors =
+                config.view( enumerate( GraphDatabaseSettings.Connector.class ) ).map( BoltConnector::new ).filter(
+                        ( connConfig ) -> BOLT.equals( config.get( connConfig.type ) ) &&
+                                config.get( connConfig.enabled ) ).map( ( connConfig ) -> {
```

Comment:
Since this code is already touched by formatting, please pull this big lambda into a method. This whole stream processing is quite hard to read otherwise.